adr/ADR-50.md

```diff
@@ -91,7 +90,82 @@ The `LastMsgId` header is currently not supported. A batch will be rejected if t
 
 Initial release of this feature rejects the use of `MsgId`. Starting from 2.12.1 de-duplication is supported and a batch will be rejected with an error if it contains a duplicate message.
 
-### Publish Acknowledgements
+## Fast-ingest Batch Publishing
+
+### Context
+
+Today we have Async publishing in the clients that aims to move data into a stream at high speed, it works well but pays a very high cost for managing acknowledgements both in client and server.
+
+We would like to build on the batching behaviours introduced here but deliver high performance ingest without atomicity.
+
+Crucially these batches are not pre-staged and so will not be limited to 1000 messages like Atomic ones.
+
+### Control Channel
+
+The heart of this design is control channel that is open and kept open for the duration of the batch. In practise this the reply subject of the first message will be used for all signaling during the batch.
+
+The server will send acks over the channel on a frequency like once every 10 messages or once every 5MB. Crucially if at any stage an error is encountered errors can be sent back immediately and received by the client.
+
+Clients will not wait for each ack like they would in a standard JS Publish instead they will maintain a count of maximum outstanding acknowledgements from the server, this is part flow-control and part outage detection.
+
+The server on the other hand will be able to adjust the frequency of acks based on internal metrics of the stream such as the size of the in-flight RAFT proposals. In this way the server and clients will balance speed of fast ingest vs responsiveness of interactive commands requiring RAFT proposals.
```

Comment:
If client can set `max ack pending` while server can adjust it depending on stream it means we have two variables that affect each other and client behavior becomes quite unpredictable. The same `max ack pending` can result in quite different outcomes. It might be fine, but I would not expose this setting to the user on the client end.